persistence-cassandra/core/src/main/scala/com/lightbend/lagom/internal/persistence/cassandra/CassandraKeyspaceConfig.scala

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <https://www.lightbend.com>
+ */
+package com.lightbend.lagom.internal.persistence.cassandra
+
+import akka.event.LoggingAdapter
+import com.typesafe.config.Config
+
+private[lagom] object CassandraKeyspaceConfig {
+
+  def validateKeyspace(namespace: String)(implicit config: Config, log: LoggingAdapter): Unit =
+    validateKeyspace(namespace, s"$namespace.defaults")
+
+  def validateKeyspace(namespace: String, defaultNamespace: String)(implicit config: Config, log: LoggingAdapter): Unit = {
+    val path = s"$namespace.keyspace"
+    if (!config.hasPath(path)) {
+      log.error("Configuration for [{}] must be set in application.conf", path)
+    } else if (log.isWarningEnabled) {
+      val keyspace = config.getString(path)
+      val defaultKeyspace = config.getString(s"$defaultNamespace.keyspace")
+      if (keyspace == defaultKeyspace) {
```

Comment:
A drawback of this approach is that this still warns if you have explicitly set the keyspace to the same value as the default.

Unfortunately, there's no way to tell the difference at this point.